src/transforms/log_to_metric.rs

```diff
@@ -35,6 +48,35 @@ const ORIGIN_SERVICE_VALUE: u32 = 3;
 pub struct LogToMetricConfig {
     /// A list of metrics to generate.
     pub metrics: Vec<MetricConfig>,
+    /// Setting this flag will change the behavior of this transformation.
+    /// Notably the `metrics` field will be ignored. 
+    /// All incoming events will be processed and if possible they will be converted to log events. 
+    /// Otherwise, only items specified in the 'metrics' field will be processes.
+    ///
+    /// ```
+    /// use serde_json::json;
+    /// let json_event = json!({
+    ///     "counter": {
+    ///         "value": 10.0
+    ///     },
+    ///     "kind": "incremental",
+    ///     "name": "test.transform.counter",
+    ///     "tags": {
+    ///         "env": "test_env",
+    ///         "host": "localhost"
+    ///     }
+    /// });
+    /// ```
+    ///
+    /// This is an example JSON representation of a counter with the following properties:
+    ///
+    /// - `counter`: An object with a single property `value` representing the counter value (e.g., 10.0).
+    /// - `kind`: A string indicating the kind of counter, in this case, "incremental".
+    /// - `name`: A string representing the name of the counter, here set to "test.transform.counter".
+    /// - `tags`: An object containing additional tags such as "env" and "host".
+    ///
+    /// Objects that can be processes include counter, histogram, gauge, set and summary.
```

Comment:
```suggestion
    /// Objects that can be processed include counter, histogram, gauge, set and summary.
```
Again 'processed' or 'processes'?